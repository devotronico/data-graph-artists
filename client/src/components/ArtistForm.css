/*
    This styling is heavily inspired by the Wikipedia Map project by Luke Deen Taylor

    Site: luke.deentaylor.com/wikipedia/
    Github: https://github.com/controversial/wikipedia-map
*/

#form-box {
    padding: 0;
    display: block;
    overflow: hidden;
  
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    z-index: 1;
  
    box-shadow: 0 4px 4px -2px #ccc;
}
  
#form-box button {
    background-color: rgba(255, 255, 255, 0.6);
  
    /* Border */
    outline: none;
  
    padding: 5px 7px;
    margin: 0;
    font-size: 18px;
    float: right;
    height: 40px;
    min-width: 33px;
}

#search-field {
    display: flex;
    flex-wrap: wrap;

    overflow: hidden;
    height: 33px;
    padding: 3px;
    margin: 0;

    font-family: "Helvetica Neue", Helvetica, sans-serif;
}

#search-field input {
    font-size: 1em;

    /* Kill default styles */
    outline: none;
    border: none;
    box-shadow: none;
    -webkit-appearance: none;
    background-color: transparent;

    margin: 2px 5px 3px 0;
    padding: 0 5px;
    flex: 2;
    height: 95%;
}
  
.transparent-blur {
    background-color: rgba(255, 255, 255, 0.65);
    -webkit-backdrop-filter: saturate(180%) blur(20px);
}