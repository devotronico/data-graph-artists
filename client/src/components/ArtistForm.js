import React from 'react';
import './ArtistForm.css';
import PropTypes from 'prop-types';

class ArtistForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleReset = this.handleReset.bind(this);
  }

  handleChange(event) {
    const { onArtistChange } = this.props;
    onArtistChange(event.target.value);
  }

  handleSubmit(event) {
    const { onArtistSubmit, searchValue } = this.props;
    onArtistSubmit(searchValue);
    event.preventDefault();
  }

  handleReset(event) {
    const { onGraphReset } = this.props;
    onGraphReset();
    event.preventDefault();
  }

  render() {
    const { searchValue } = this.props;
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <form id="form-box" className="transparent-blur" onSubmit={this.handleSubmit}>
        <button type="button" id="graph-reset" onClick={this.handleReset}>
          Reset
        </button>
        <button type="button" id="artist-submit" onClick={this.handleSubmit}>
          Search
        </button>
        <div id="search-field">
          <input
            type="text"
            value={searchValue}
            onChange={this.handleChange}
            placeholder="Artist Name"
          />
        </div>
      </form>
    );
  }
}

ArtistForm.propTypes = {
  searchValue: PropTypes.string,
  onArtistChange: PropTypes.func,
  onArtistSubmit: PropTypes.func,
  onGraphReset: PropTypes.func,
};

ArtistForm.defaultProps = {
  searchValue: '',
  onArtistChange: () => {},
  onArtistSubmit: () => {},
  onGraphReset: () => {},
};

export default ArtistForm;
