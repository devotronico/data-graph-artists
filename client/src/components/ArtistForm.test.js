import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ArtistForm from './ArtistForm';

configure({ adapter: new Adapter() });

describe('ArtistForm', () => {
  it('renders correctly', () => {
    const mockHandleArtistChange = jest.fn();
    const mockHandleArtistSubmit = jest.fn();
    const mockHandleGraphReset = jest.fn();
    const component = renderer.create(
      <ArtistForm
        searchValue=""
        onArtistChange={mockHandleArtistChange}
        onArtistSubmit={mockHandleArtistSubmit}
        onGraphReset={mockHandleGraphReset}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the artist change callback function when input text changes', () => {
    const mockHandleArtistChange = jest.fn();
    const mockHandleArtistSubmit = jest.fn();
    const mockHandleGraphReset = jest.fn();
    const event = {
      target: {
        value: 'test',
      },
    };

    const component = shallow(
      <ArtistForm
        searchValue=""
        onArtistChange={mockHandleArtistChange}
        onArtistSubmit={mockHandleArtistSubmit}
        onGraphReset={mockHandleGraphReset}
      />,
    );

    component.find('input').simulate('change', event);
    expect(mockHandleArtistChange).toHaveBeenCalledWith('test');
  });

  it('calls the artist submit callback function when input is submitted', () => {
    const mockHandleArtistChange = jest.fn();
    const mockHandleArtistSubmit = jest.fn();
    const mockHandleGraphReset = jest.fn();

    const component = shallow(
      <ArtistForm
        searchValue=""
        onArtistChange={mockHandleArtistChange}
        onArtistSubmit={mockHandleArtistSubmit}
        onGraphReset={mockHandleGraphReset}
      />,
    );

    component.find('#artist-submit').simulate('click', {
      preventDefault: () => {},
    });
    expect(mockHandleArtistSubmit).toHaveBeenCalled();
  });

  it('calls the graph reset callback function when the reset button is clicked', () => {
    const mockHandleArtistChange = jest.fn();
    const mockHandleArtistSubmit = jest.fn();
    const mockHandleGraphReset = jest.fn();

    const component = shallow(
      <ArtistForm
        searchValue=""
        onArtistChange={mockHandleArtistChange}
        onArtistSubmit={mockHandleArtistSubmit}
        onGraphReset={mockHandleGraphReset}
      />,
    );

    component.find('#graph-reset').simulate('click', {
      preventDefault: () => {},
    });
    expect(mockHandleGraphReset).toHaveBeenCalled();
  });
});
