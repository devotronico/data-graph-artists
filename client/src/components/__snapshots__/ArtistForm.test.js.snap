// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistForm renders correctly 1`] = `
<form
  className="transparent-blur"
  id="form-box"
  onSubmit={[Function]}
>
  <button
    id="graph-reset"
    onClick={[Function]}
    type="button"
  >
    Reset
  </button>
  <button
    id="artist-submit"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
  <div
    id="search-field"
  >
    <input
      onChange={[Function]}
      placeholder="Artist Name"
      type="text"
      value=""
    />
  </div>
</form>
`;
